src/librustc_codegen_llvm/builder.rs

```diff
@@ -1068,6 +1069,20 @@ impl StaticBuilderMethods for Builder<'a, 'll, 'tcx> {
         self.cx().get_static(def_id)
     }
 
+    fn static_panic_location(&mut self, loc: &syntax::source_map::Loc, kind: &str) -> Self::Value {
+        let filename = Symbol::intern(&loc.file.name.to_string());
+        let filename = self.const_str(filename);
+        let line = self.const_u32(loc.line as u32);
+        let col = self.const_u32(loc.col.to_usize() as u32 + 1);
+        let struct_ = self.const_struct(&[filename.0, filename.1, line, col], false);
+
+        let align = self.tcx.data_layout.aggregate_align.abi
+            .max(self.tcx.data_layout.i32_align.abi)
+            .max(self.tcx.data_layout.pointer_align.abi);
+        // FIXME(eddyb) move this into miri, it can be correct if e.g. field order changes
+        self.static_addr_of(struct_, align, Some(kind))
```

Comment:
You can hardcode the name here ("kind" is inaccurate, it's really just a global name), or not use one at all.